src/main/java/knn/api/APIKnowledgeNode.java

```diff
@@ -0,0 +1,101 @@
+package knn.api;
+
+import prometheus.api.GeneralInputAPI;
+import tags.Tag;
+
+import java.util.Set;
+
+/**
+ * APIKnowledgeNodeInterface.
+ */
+public abstract class APIKnowledgeNode implements GeneralInputAPI {
```

Comment:
```suggestion
public abstract class APIKnowledgeNode implements GeneralInputAPI, GeneralOutputAPI{
```
From what I understood from the prof's notes he wants the API to implement both of those interfaces. 

This was from the document

class KnowledgeNode implements LayerInput, LayerOutput

We could also rename GeneralInputAPI with LayerInput same thing for output. Lmk what you think.